hydrolib/hydromt_delft3dfm/workflows/branches.py

```diff
@@ -455,50 +486,73 @@ def _split_branches_by_spacing_const(
     edge_bedlevdn = []
     edge_index = []
     branch_index = []
-    for bid, b in branches.iterrows():
-        import pdb
 
-        pdb.set_trace()
+    # Check for attributes
+    interp_invlev = False
+    interp_bedlev = False
+    if "invlev_up" and "invlev_dn" in branches.columns:
+        interp_invlev = True
+    if "bedlev_up" and "bedlev_dn" in branches.columns:
+        interp_bedlev = True
+
+    for bid, b in branches.iterrows():
         # prepare for splitting
         line = b.geometry
         num_new_lines = int(np.ceil(line.length / spacing_const))
 
-        # interpolate geometry
-        new_edges = split_lines(line, num_new_lines)
-        offsets = np.linspace(0, line.length, num_new_lines + 1)
-
-        # interpolate values
-        edge_geom.extend(new_edges)
-        edge_offset.extend(offsets[1:])
-        edge_invertup.extend(
-            np.interp(
-                offsets[:-1], [0, offsets[-1]], [b.invlev_up, b.invlev_dn]
-            )  # TODO: renaming needed
-        )
-        edge_invertdn.extend(
-            np.interp(offsets[1:], [0, offsets[-1]], [b.invlev_up, b.invlev_dn])
-        )
-        edge_bedlevup.extend(
-            np.interp(offsets[:-1], [0, offsets[-1]], [b.bedlev_up, b.bedlev_dn])
-        )
-        edge_bedlevdn.extend(
-            np.interp(offsets[1:], [0, offsets[-1]], [b.bedlev_up, b.bedlev_dn])
-        )
-        edge_index.extend([bid + "_E" + str(i) for i in range(len(new_edges))])
-        branch_index.extend([bid] * len(new_edges))
+        if num_new_lines > 0:
```

Comment:
Doing the following will make the code less nested 

```
        if num_new_lines <= 0:
            continue

        # interpolate geometry
            new_edges = split_lines(line, num_new_lines)
            if smooth_branches:
                ...
            ...
```

Reply:
Check